worker/include/Utils.hpp

```diff
@@ -377,18 +418,33 @@ namespace Utils
 			return ptr;
 		}
 
-		// Return allocated memory into internal pool for future use, make sure to run destructor before
-		// returning memory, ObjectPool will only de-allocate memory on exit.
+		void deallocate(T* ptr, size_t n)
+		{
+			if (n > 1 && ptr)
+			{
+				std::free(ptr);
+				return;
+			}
+
+			this->Return(ptr);
+		}
+
 		void Return(T* ptr)
 		{
-			if (ptr)
-				this->pool.push_back(ptr);
+			if (!ptr)
+			{
+				return;
+			}
+
+			this->pool.push_back(ptr);
 		}
```

Comment:
Maybe we don't need `Allocate` and `Return` and can instead rely on standard methods from `std::allocator` API?